designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Button.kt

```diff
@@ -0,0 +1,180 @@
+                }
+
+                Text(text, color = textColor, textAlign = TextAlign.Center)
+            }
+        }
+        ButtonStyle.ELEVATED -> {
+            val textColor = if (enabled) SurfaceColor.Primary else TextColor.OnDisabledSurface
+
+            ElevatedButton(
+                onClick = { onClick() },
+                modifier = Modifier,
+                elevation = ButtonDefaults.elevatedButtonElevation(),
+                enabled = enabled,
+                colors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.ContainerLow, SurfaceColor.Primary, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                shape = ButtonDefaults.outlinedShape,
+                contentPadding = paddingValues
+            ) {
+                if (icon != null) {
+                    icon()
+                    Spacer(Modifier.size(Spacing.Spacing8))
+                }
+
+                Text(text, color = textColor, textAlign = TextAlign.Center)
+            }
+        }
+        ButtonStyle.TONAL -> {
+            val textColor = if (enabled) TextColor.OnPrimaryContainer else TextColor.OnDisabledSurface
+            CompositionLocalProvider(LocalRippleTheme provides Ripple.CustomDHISRippleTheme) {
+                Button(
+                    onClick = { onClick() },
+                    modifier = Modifier,
+                    enabled = enabled,
+                    colors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.PrimaryContainer, TextColor.OnPrimaryContainer, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                    shape = ButtonDefaults.outlinedShape,
+                    contentPadding = paddingValues
+                ) {
+                    if (icon != null) {
+                        icon()
+                        Spacer(Modifier.size(Spacing.Spacing8))
+                    }
+
+                    Text(text, color = textColor, textAlign = TextAlign.Center)
+                }
+            }
+        }
+        ButtonStyle.KEYBOARDKEY -> {
+            val textColor = if (enabled) SurfaceColor.Primary else TextColor.OnDisabledSurface
+            val shadowColor = if (enabled) SurfaceColor.ContainerHighest else Color.Transparent
+            Button(
+                onClick = { onClick() },
+                modifier = Modifier.buttonShadow(shadowColor, Radius.Full, icon != null),
+                enabled = enabled,
+                colors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.Container, SurfaceColor.Primary, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                shape = ButtonDefaults.outlinedShape,
+                contentPadding = paddingValues
+            ) {
+                if (icon != null) {
+                    icon()
+                    Spacer(Modifier.size(Spacing.Spacing8))
+                }
+
+                Text(text, color = textColor, textAlign = TextAlign.Center)
+            }
+        }
+        else -> {
```

Comment:
Maybe this when can be exhaustive, so that, if we add another value, this will trigger an error and force us to implement.